CaesarCipher.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package caesarcipher;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author kailabillie
+ */
+public class CaesarCipher {
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+
+        String encodedString = Encode("-1:some text");
+        System.out.println(encodedString);
+
+    }
+
+    static String Encode(String word) {
+
+        String uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        String lowercase = "abcdefghijklmnopqrstuvwxyz";
+        String digits = "1234567890";
+        
+        String [] words = word.split(":");
+        
+        
+        
+        int letterShift = Integer.parseInt(words[0]);
+        String encodeWord = words[1];
+        
+        System.out.println(encodeWord);
+        
+        String newWord = "";
+        
+        for (int i = 0; i<encodeWord.length(); i++) {
+            if (String.valueOf(encodeWord.charAt(i)).equals(" ")) {
```

Comment:
You could choose to utilize an else statement to handle adding all special characters 'as is' to the encoded string by removing lines 46-48, and adding this else statement after your last if condition. This will make it pass special characters test cases.
```
else {
     newWord.append(encodeWord.charAt(i));
}
```